### PR TITLE
[clang][NFC] Reformat suspicious condition

### LINKED ISSUE #89805: clang/lib/Lex/Pragma.cpp:1447:14: style: Suspicious condition
Static analyser cppcheck says:

clang/lib/Lex/Pragma.cpp:1447:14: style: Suspicious condition (assignment + comparison); Clarify expression with parentheses. [clarifyCondition]

Source code is

         if ((SpecifierValid = SpecifierInt != -1))

I suspect the following code was intended:

         if (SpecifierValid = (SpecifierInt != -1))

### PR DESCRIPTION
**Pull request by Troy-Butler**
Addresses issue #89805.
Assignment + comparison performed in conditional statement. Resolved by parenthesizing comparison operation.

Fixes #89805.

### CODE REVIEW

**Review by cor3ntin — APPROVED** [review]
Thanks for your contribution.
Will you need me to merge that for you?

**Comment by Troy-Butler** [comment]
> Thanks for your contribution. Will you need me to merge that for you?

Yes please! I don't have merging rights. Thank you!

**Review by shafik — COMMENTED** [review]
This makes sense, I added Nico since they added the change that brought in that line.

**Comment by Troy-Butler** [comment]
I see that the build has failed - what do I need to do to fix this?

**Comment by shafik** [comment]
The `amdgpu-toolchain.c` test failure looks unrelated. I think we need another empty commit to kick off the build again unfortunately.

**Comment by cor3ntin** [comment]
I think CI is having some difficulties. No need to wait for it for this.
Thanks a lot for your first contribution!
We have some good first issues if you want to have a go at some more contributions
https://github.com/llvm/llvm-project/issues?q=is%3Aissue+is%3Aopen+good+first+issues+label%3A%22good+first+issue%22

File: clang/lib/Lex/Pragma.cpp [hunk 1]
```diff
@@ -1444,7 +1444,7 @@ struct PragmaWarningHandler : public PragmaHandler {
                                  .Case("once", PPCallbacks::PWS_Once)
                                  .Case("suppress", PPCallbacks::PWS_Suppress)
                                  .Default(-1);
-          if ((SpecifierValid = SpecifierInt != -1))
+          if (SpecifierValid = (SpecifierInt != -1))
```
Comment (fhahn) [inline clang/lib/Lex/Pragma.cpp:1447]: Unfortunately this introduces a new warning

```
/Users/florianhahn/projects/llvm-project/clang/lib/Lex/Pragma.cpp:1447:30: warning: using the result of an assignment as a condition without parentheses [-Wparentheses]
 1447 |           if (SpecifierValid = (SpecifierInt != -1))
      |               ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
/Users/florianhahn/projects/llvm-project/clang/lib/Lex/Pragma.cpp:1447:30: note: place parentheses around the assignment to silence this warning
 1447 |           if (SpecifierValid = (SpecifierInt != -1))
      |                              ^
      |               (                                    )
/Users/florianhahn/projects/llvm-project/clang/lib/Lex/Pragma.cpp:1447:30: note: use '==' to turn this assignment into an equality comparison
 1447 |           if (SpecifierValid = (SpecifierInt != -1))
      |                              ^
      |                              ==
1 warning generated.
```

We likely need to outer parenthesis as well

```
diff --git a/clang/lib/Lex/Pragma.cpp b/clang/lib/Lex/Pragma.cpp
index ced407355e00..2972ee2197e2 100644
--- a/clang/lib/Lex/Pragma.cpp
+++ b/clang/lib/Lex/Pragma.cpp
@@ -1444,7 +1444,7 @@ struct PragmaWarningHandler : public PragmaHandler {
                                  .Case("once", PPCallbacks::PWS_Once)
                                  .Case("suppress", PPCallbacks::PWS_Suppress)
                                  .Default(-1);
-          if (SpecifierValid = (SpecifierInt != -1))
+          if ((SpecifierValid = (SpecifierInt != -1)))
             Specifier =
                 static_cast<PPCallbacks::PragmaWarningSpecifier>(SpecifierInt);
```

**Pull request merged**